src/LayerPlan.cpp

```diff
@@ -749,9 +754,30 @@ void LayerPlan::addWall(ConstPolygonRef wall, int start_idx, const GCodePathConf
             computeDistanceToBridgeStart((start_idx + point_idx - 1) % wall.size());
         }
 
-        addWallLine(p0, p1, non_bridge_config, bridge_config, flow, non_bridge_line_volume, speed_factor, distance_to_bridge_start);
+        // HACK ALERT
+        // the overlap compensation is not perfect, it can produce short non-flow reduced line segments within a sequence of flow reduced
+        // line segments and so to try and avoid printing the spurious fat line segments we require that their lengths are above a threshold
+
+        const int64_t max_spurious_fat_segment_length = 50; // microns
```

Comment:
Some notes about this: According to Ruben, this should be using coordinates, as well as the global resolution parameters.

We don't think this hack is too bad but are wondering what the consequences of not having it would be...

Reply:
> Some notes about this: According to Ruben, this should be using coordinates, as well as the global resolution parameters.

Not sure I understand what you mean by using coordinates. As for the global resolution params, I don't think they are accessible in this method, are they?

> We don't think this hack is too bad but are wondering what the consequences of not having it would be...

If you don't have the hack then tiny line segments can be included which are not really wanted. Let me try and come up with a concrete example...

Reply:
I can't recreate the situation that made me want to filter out the tiny fat line segments right now but I am sure that I added that because I was seeing a problem. I don't think I just put it in from the beginning in the hope that it would be useful.

Reply:
@Ghostkeeper

Reply:
> Not sure I understand what you mean by using coordinates.

I mean it should be a `coord_t`

Makes sense regarding the line segments. For me it's OK because it's logical, but was still worth asking.

Reply:
Ah, understood. I've made that little change.